client/state/login/actions/reboot-after-login.ts

```diff
@@ -24,11 +24,11 @@ export const rebootAfterLogin = ( tracksEventArgs: Record< string, unknown > ) =
 	// Redirects to / if no redirect url is available
 	const url = getRedirectToSanitized( getState() ) || '/';
 
-	// user data is persisted in localstorage at `lib/user/user` line 157
+	// user ID is persisted in localstorage
 	// therefore we need to reset it before we redirect, otherwise we'll get
 	// mixed data from old and new user
-	if ( user().get() ) {
-		await user().clear();
+	if ( getStoredUserId() ) {
+		await clearStore();
```

Comment:
I think it wouldn't be bad to always call `clearStore`, without the `getStoredUserId` check. After a fresh login, we always want to start with a blank slate.

Reply:
Yeah, I think you're right. I'm happy to follow up in another PR.